src/test/scala/com/databricks/spark/avro/AvroSuite.scala

```diff
@@ -544,4 +544,16 @@ class AvroSuite extends FunSuite with BeforeAndAfterAll {
 
     assert(withEmptyColumn.forall(_ == Row(null: String, Row(null: String, null: String))))
   }
+
+  test("read avro file partitioned") {
+    TestUtils.withTempDir { dir =>
+      val sparkSession = spark
+      import sparkSession.implicits._
+      val df = (0 to 1024 * 3).toDS.map(i => s"record${i}").toDF("records")
+      val outputDir = s"$dir/${UUID.randomUUID}"
+      df.write.avro(outputDir)
+      val input = spark.read.avro(outputDir)
+      assert(input.collect.toSet.size === 1024 * 3 + 1)
```

Comment:
This tests that the read record count is the same as the written record count, but not that the read Dataset is split into multiple partitions.

Can you add something like this that would confirm that as well?

`assert(input.rdd.partitions.length > 1)`